subsys/pm/device_runtime.c

```diff
@@ -83,15 +90,19 @@ static void runtime_suspend_work(struct k_work *work)
 	int ret;
 	struct pm_device *pm = CONTAINER_OF(work, struct pm_device, work);
 
-	ret = pm->action_cb(pm->dev, PM_DEVICE_ACTION_SUSPEND);
-
 	(void)k_mutex_lock(&pm->lock, K_FOREVER);
-	if (ret == 0) {
-		pm->state = PM_DEVICE_STATE_SUSPENDED;
-	}
+	ret = pm_device_action_run(pm->dev, PM_DEVICE_ACTION_SUSPEND);
```

Comment:
note: I assume this could be left as it was now?

Reply:
I believe so, but I'm not entirely clear why it was changed in the first place